packages/server/graphql/mutations/deleteUser.ts

```diff
@@ -43,43 +54,24 @@ export default {
     }
     const su = isSuperUser(authToken)
     const viewerId = getUserId(authToken)
-
     const user = userId ? await getUserById(userId) : email ? await getUserByEmail(email) : null
+    const validReason = reason?.trim().slice(0, USER_REASON_REMOVED_LIMIT) || 'No reason provided'
+
     if (!su) {
       if (!user || userId !== viewerId) {
         return {error: {message: 'Cannot delete someone else'}}
       }
     } else if (!user) {
       return {error: {message: 'User not found'}}
     }
-    const {id: userIdToDelete, tms} = user
-    removeSlackAuths(userIdToDelete, tms, true)
-    const orgUsers = await dataLoader.get('organizationUsersByUserId').load(userIdToDelete)
-    const orgIds = orgUsers.map((orgUser) => orgUser.orgId)
-    await Promise.all(
-      orgIds.map((orgId) => removeFromOrg(userIdToDelete, orgId, undefined, dataLoader))
-    )
-    const validReason = reason?.trim().slice(0, USER_REASON_REMOVED_LIMIT) || 'No reason provided'
-    if (userId) {
-      segmentIo.track({
-        userId,
-        event: 'Account Removed',
-        properties: {
-          reason: validReason
-        }
-      })
-    }
+    const {id: userIdToDelete} = user
+
+    await softDeleteUserResolver(userIdToDelete, dataLoader, validReason)
+
     // do this after 30 seconds so any segment API calls can still get the email
-    const update = {
-      isRemoved: true,
-      email: getDeletedEmail(userId),
-      reasonRemoved: validReason,
-      updatedAt: new Date()
-    }
-    setTimeout(() => {
-      db.write('User', userIdToDelete, update)
-      updateUser(update, userIdToDelete)
-    }, 30000)
+    await delay(30000)
```

Comment:
before the long sleep was fine because nothing in our system was really impacted-- we already returned from the resolver & we were just waiting 30 seconds so integrations could still grab the email of the to-be-deleted user.

now that we have 1 function waiting on sleep, we should do this more elegantly.

what we need to do is share the following JSON object with hubspot (https://github.com/ParabolInc/parabol/blob/439d988af10717e2de9a9443c29185f2f5e6b81f/packages/server/utils/updateHubspot.ts#L108-L118)

to do that:
- pass in the referenced string as a query with the userId as the variables to graphql. it should resolve that json object
- next, we pass that object into the properties field here: https://github.com/ParabolInc/parabol/blob/9dbdda9f39b99b0db9b4bf77cbc92557187f0a4c/packages/server/graphql/mutations/helpers/softDeleteUser.ts#L46
- finally, we write a custom conditional for that event to use the data in the `properties` field instead of reaching back out to our endpoint: https://github.com/ParabolInc/parabol/blob/439d988af10717e2de9a9443c29185f2f5e6b81f/packages/server/utils/updateHubspot.ts#L384
the end result is removing the `delay(30000)` 🎉